### PR TITLE
Add format output option

### PR DESCRIPTION
**Pull request by AyrtonRicardo**
- Add test
- Add format output option

### CODE REVIEW

**Comment by goetas** [comment]
this is a XML specific option, and should not be part of the global context.

have a look to https://github.com/schmittjoh/serializer/blob/master/src/JMS/Serializer/JsonSerializationVisitor.php#L48 for an alternative implementation.

https://github.com/schmittjoh/serializer/commit/4be3f290903643caac5e238d3f7bdd6b3950d0c5 added a similar feature for JSON

**Comment by AyrtonRicardo** [comment]
@goetas I agree with you.

Done.

**Comment by goetas** [comment]
looks good to me! :+1:

**Comment by AyrtonRicardo** [comment]
@goetas done schmittjoh/JMSSerializerBundle#527

File: src/JMS/Serializer/XmlSerializationVisitor.php [hunk 1]
```diff
@@ -44,10 +44,14 @@ class XmlSerializationVisitor extends AbstractVisitor
     private $nullWasVisited;
     private $objectMetadataStack;
 
+    /** @var boolean */
+    private $formatOutput;
```
Comment (goetas) [inline src/JMS/Serializer/XmlSerializationVisitor.php]: set it default to false, please

we had some bugs fro such kind of implementation (https://github.com/schmittjoh/serializer/pull/317)

Comment (AyrtonRicardo) [inline src/JMS/Serializer/XmlSerializationVisitor.php]: It's default `true`, setted on constructor. It's default `true` because it was `true` before.

**Pull request merged**